Symlinks support in copy tasks (#25264)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

> Get more help at https://help.gradle.org.
```

<a name="symlinks"></a>

### Symlinks support in Copy tasks

> One of the core principles of Gradle was to treat any symbolic link as a file it points to.

Reviewer: ```suggestion
One of the core principles of Gradle is to treat a symbolic link as the file it points to.
```

However, there are some cases when this approach is not desirable.
One of the common use cases is packing or unpacking archives with symlinks: symlinks help reduce archive size and speed up the packing process.

> The same can be said about copying files with symlinks: it is a lot faster to copy a symlink than a file or a directory.

Reviewer: ```suggestion
The same can be said about copying files with symlinks: copying a symlink is much faster than a file or a directory.
```

This release presents a new property to [`CopySpec`](javadoc/org/gradle/api/file/CopySpec.html): `linksStrategy`.
Using this property, you can control how symlinks are handled during the copy process.

> The default value of `linksStrategy` for a regular copy task is `LinksStrategy.FOLLOW`, which preserves the existing behavior of Gradle of following every symlink.

Reviewer: ```suggestion
The default value of `linksStrategy` for a regular copy task is `LinksStrategy.FOLLOW`, which preserves the existing behavior of Gradle, following every symlink.
```

If the source file is an archive ([`zipTree`](javadoc/org/gradle/api/file/ArchiveOperations.html#zipTree-java.lang.Object-)

> or [`tarTree`](javadoc/org/gradle/api/file/ArchiveOperations.html#tarTree-java.lang.Object-), to be more specific), then the default value is `LinksStrategy.PRESERVE_RELATIVE`.

Reviewer: ```suggestion
or [`tarTree`](javadoc/org/gradle/api/file/ArchiveOperations.html#tarTree-java.lang.Object-)), then the default value is `LinksStrategy.PRESERVE_RELATIVE`.
```

With this strategy, symlinks from the archive are preserved unless they are pointing to a location outside the archive.

> Other possible strategies are `LinksStrategy.ERROR` to fail on any symlink and `LinksStrategy.PRESERVE_ALL` which preserves all symlinks.

Reviewer: ```suggestion
`LinksStrategy.ERROR` fails on any symlink, and `LinksStrategy.PRESERVE_ALL` preserves all symlinks.
```
